provider/kubernetes/kubernetes_test.go

```diff
@@ -2262,6 +2263,297 @@ func TestBasicAuthInTemplate(t *testing.T) {
+				ClusterIP: "10.0.0.1",
+				Type:      "ClusterIP",
+				Ports: []v1.ServicePort{
+					{
+						Name: "http",
+						Port: 80,
+					},
+				},
+			},
+		},
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "example-org",
+				UID:       "1",
+				Namespace: "testing",
+			},
+			Spec: v1.ServiceSpec{
+				ClusterIP: "10.0.0.2",
+				Type:      "ClusterIP",
+				Ports: []v1.ServicePort{
+					{
+						Name: "http",
+						Port: 80,
+					},
+				},
+			},
+		},
+	}
+	secrets := []*v1.Secret{
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "myTlsSecret",
+				UID:       "1",
+				Namespace: "testing",
+			},
+			Data: map[string][]byte{
+				"tls.crt": []byte("-----BEGIN CERTIFICATE-----\n-----END CERTIFICATE-----"),
+				"tls.key": []byte("-----BEGIN PRIVATE KEY-----\n-----END PRIVATE KEY-----"),
+			},
+		},
+	}
+	endpoints := []*v1.Endpoints{}
+	watchChan := make(chan interface{})
+	client := clientMock{
+		ingresses: ingresses,
+		services:  services,
+		secrets:   secrets,
+		endpoints: endpoints,
+		watchChan: watchChan,
+	}
+	provider := Provider{}
+	actual, err := provider.loadIngresses(client)
+	if err != nil {
+		t.Fatalf("error %+v", err)
+	}
+
+	expected := &types.Configuration{
+		Backends: map[string]*types.Backend{
+			"example.com": {
+				Servers:        map[string]types.Server{},
+				CircuitBreaker: nil,
+				LoadBalancer: &types.LoadBalancer{
+					Method: "wrr",
+				},
+			},
+			"example.org": {
+				Servers:        map[string]types.Server{},
+				CircuitBreaker: nil,
+				LoadBalancer: &types.LoadBalancer{
+					Method: "wrr",
+				},
+			},
+		},
+		Frontends: map[string]*types.Frontend{
+			"example.com": {
+				Backend:        "example.com",
+				PassHostHeader: true,
+				Routes: map[string]types.Route{
+					"example.com": {
+						Rule: "Host:example.com",
+					},
+				},
+			},
+			"example.org": {
+				Backend:        "example.org",
+				PassHostHeader: true,
+				Routes: map[string]types.Route{
+					"example.org": {
+						Rule: "Host:example.org",
+					},
+				},
+			},
+		},
+		TLSConfiguration: []*tls.Configuration{
+			{
+				EntryPoints: []string{"example.com"},
+				Certificate: &tls.Certificate{
+					CertFile: "-----BEGIN CERTIFICATE-----\n-----END CERTIFICATE-----",
+					KeyFile:  "-----BEGIN PRIVATE KEY-----\n-----END PRIVATE KEY-----",
+				},
+			},
+		},
+	}
+
+	assert.Equal(t, expected, actual)
+}
+
+func TestGetTLSConfigurations(t *testing.T) {
+	testIngress := v1beta1.Ingress{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "test-ingress",
+			Namespace: "testing",
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "ep1.example.com",
+				},
+				{
+					Host: "ep2.example.com",
+				},
+			},
+			TLS: []v1beta1.IngressTLS{{SecretName: "test-secret"}},
+		},
+	}
+
+	tests := []struct {
+		desc      string
+		ingress   *v1beta1.Ingress
+		client    Client
+		result    []*tls.Configuration
+		errResult string
+	}{
+		{
+			desc:    "error_api_client",
```

Comment:
It's okay to use human-readable descriptions: The Go test package automatically translates whitespaces and other non-"computer friendly" characters.